odoo/addons/base/models/ir_model.py

```diff
@@ -1980,8 +1980,17 @@ def delete(records):
         # remove models
         delete(self.env['ir.model'].browse(model_ids))
 
+        # attempt to remove undeletable model data that may have become deletable again because
+        # of the tables being dropped just above
+        for data in self.browse(undeletable_ids).exists():
+            record = self.env[data.model].browse(data.res_id)
+            if record.exists():
+                # record exists therefore the data is still undeletable, remove it from module_data
+                module_data -= data
+            else:
+                _logger.warning(f"Found orphan ir.model.data ({data.name}) from module {data.module}")
```

Comment:
We should not warn about orphan external ids IMHO, because their corresponding record has actually been deleted (as it should have been), and they are going to be deleted (as they should).  We should instead warn about the external ids that we really cannot delete.
```suggestion
        # some undeletable records may actually have been cascade-deleted;
        # determine which ones are still there to not delete their external id
        undeletable = self.browse(undeletable_ids).exists().filtered(
            lambda data: self.env[data.model].browse(data.res_id).exists()
        )
        for data in undeletable:
            _logger.warning("Could not delete %s.%s", data.module, data.name)
        module_data -= undeletable
```